### PR TITLE
refactor subroutine set_ub_col and setcol

### PR DESCRIPTION
**Pull request by tangshuaiqi**
refactor out two small subroutines
clean up some unused variables
change a variable name ncol_abs to nabscol as defined in chem_mods.

### CODE REVIEW

File: components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90 [hunk 1]
```diff
@@ -563,7 +563,7 @@ subroutine gas_phase_chemdr(lchnk, ncol, imozart, q, &
     !-----------------------------------------------------------------------      
     !        ... Set the column densities at the upper boundary
     !-----------------------------------------------------------------------      
-    call set_ub_col( col_delta, vmr, invariants, pint(:,1), pdel, ncol, lchnk)
+    call set_ub_col( col_delta, vmr, invariants, pdel, ncol, lchnk)
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_gas_phase_chemdr.F90]: Adding in/outs here will be useful.

File: components/eam/src/chemistry/mozart/mo_waccm_hrates.F90 [hunk 2]
```diff
@@ -272,7 +272,7 @@ subroutine waccm_hrates(ncol, state, asdir, bot_mlt_lev, qrs_tot, pbuf )
 !-----------------------------------------------------------------------      
 !        ... set the column densities at the upper boundary
 !-----------------------------------------------------------------------      
-      call set_ub_col( col_delta, vmr, invariants, state%pint(:,1), state%pdel, ncol, lchnk )
+      call set_ub_col( col_delta, vmr, invariants, state%pdel, ncol, lchnk )
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/mozart/mo_waccm_hrates.F90]: Adding in/outs here will be useful.

**Review by singhbalwinder — APPROVED** [review]
Looks great! Factoring out two subroutines made the code quite simple and easy to follow. I have left two very minor comments.

**Comment by tangshuaiqi** [comment]
All tests passed. merge into maint

comp_testall_setcol_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**